File: backend/app/schemas/company.py
from pydantic import BaseModel, field_validator
from typing import Optional

class CompanyCreate(BaseModel):
    name: str
    tax_number: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    address: Optional[str] = None
    is_active: bool = True

    @field_validator("name")
    @classmethod
    def name_must_not_be_empty(cls, value: str) -> str:
        if not value.strip():
            raise ValueError("Company name must not be empty")
        return value
    
class CompanyUpdate(BaseModel):
    name: Optional[str] = None
    tax_number: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    address: Optional[str] = None
    is_active: Optional[bool] = None

class CompanyActivate(BaseModel):
    is_active: bool

class Company(BaseModel):
    id: int
    name: str
    tax_number: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    address: Optional[str] = None
    is_active: bool

    class Config:
        from_attributes = True

File: frontend/lib/data/models/company.dart
class Company {
  const Company({
    required this.id,
    required this.name,
    this.taxNumber,
    this.phone,
    this.email,
    this.address,
    required this.isActive,
  });

  final int id;
  final String name;
  final String? taxNumber;
  final String? phone;
  final String? email;
  final String? address;
  final bool isActive;

  factory Company.fromJson(Map<String, dynamic> json) {
    return Company(
      id: _asInt(json['id']),
      name: (json['name'] ?? '').toString(),
      taxNumber: json['tax_number']?.toString(),
      phone: json['phone']?.toString(),
      email: json['email']?.toString(),
      address: json['address']?.toString(),
      isActive: json['is_active'] != false,
    );
  }

  static int _asInt(Object? value) {
    if (value is int) return value;
    if (value is num) return value.toInt();
    return int.tryParse(value?.toString() ?? '') ?? 0;
  }
}
